### PR TITLE
Add 'Test Networks Enabled' option in settings

### LINKED ISSUE #17826: Wallet: Add testnet setting for new wallet
The Testnet feature needs to be added for the new wallet. 

In Testnet mode the behaviour is exactly the same, except the UI will have an orange banner at the top saying in Testnet - reach out to design for exact details.

Please discuss behaviour with design team and desktop team about how this should behave.
e.g do all assets need to be reloaded? 
should the user basically login again?

Acceptance Criteria :
- Add this toggle in the Profile Settings page, it can be done in the old UI for the moment. 
- update relevant wallet subs to pull in the correct data for chain-id's etc.

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes: https://github.com/status-im/status-mobile/issues/17826

Add **Test Networks Enabled** toggle in Advanced Settings

### Test Notes

1. Go to Advanced settings
2. Toggle **Test Networks Enabled** option
3. The user will be logged out from the account automatically
4. Log back in & see if the **Test Networks Enabled** option persists in the previous state.

Note: This option is required to test curated communities in `status.test` fleet. This PR doesn't fix the issue logged here: https://github.com/status-im/status-mobile/issues/17852, additional changes are required to fully fix that.

### CODE REVIEW

File: src/status_im2/contexts/communities/discover/view.cljs [hunk 1]
```diff
@@ -242,6 +242,7 @@
 
 (defn- internal-discover-view
   [params]
+  (rf/dispatch [:fetch-contract-communities])
```
Comment (ajayesivan) [inline src/status_im2/contexts/communities/discover/view.cljs:245]: This dispatch is not related to the **Test Networks Enabled** option.

Just added back the curated communities fetch event dispatch that got removed in a [previous PR](https://github.com/status-im/status-mobile/pull/17094/files#diff-427a453633106ea987e59e011963a453be85a66a9cb88f34359346c35287e382L228).

**Review by jo-mut — APPROVED** [review]

File: src/status_im2/contexts/profile/settings/events.cljs [hunk 2]
```diff
@@ -56,6 +57,10 @@
             {:profile.settings/webview-debug-changed value}
             (profile-update :webview-debug (boolean value) {})))
 
+(rf/reg-event-fx :profile.settings/change-test-networks-enabled
```
Comment (J-Son89) [inline src/status_im2/contexts/profile/settings/events.cljs]: is it possible to add a simple test here? 🙏

**Review by J-Son89 — APPROVED** [review]

**Comment by J-Son89** [comment]
@briansztamfater, @vkjr, @OmarBasem I believe we can use this for wallet test-net being enabled. 

Also @ajayesivan - I believe there is some corresponding UI for this where there is an orange banner at the top of the application. Should we add this here or in a follow up?

File: src/status_im2/contexts/profile/settings/events.cljs [hunk 3]
```diff
@@ -56,6 +57,10 @@
...
+ (fn [_ [value]]
+   {:fx [[:dispatch [:profile.settings/profile-update :test-networks-enabled? (boolean value) {}]]]}))
```
Comment (ilmotta) [inline src/status_im2/contexts/profile/settings/events.cljs]: I don't see much of a need for this new event if it's just re-dispatching and passing `value` converted to a boolean.

We can create a more useful event, like `:profile.settings/toggle-test-networks` and have it read from the app db itself. So the view wouldn't even need to pass `value`, since that's already in the global state.

Or just consume `:profile.settings/profile-update` directly in the view.

```diff
modified   src/status_im/ui/screens/advanced_settings/views.cljs
@@ -101,8 +101,7 @@
      :accessibility-label :test-networks-enabled
      :container-margin-bottom 8
      :on-press
-     #(re-frame/dispatch
-       [:profile.settings/change-test-networks-enabled (not test-networks-enabled?)])
+     #(re-frame/dispatch [:profile.settings/toggle-test-networks])
      :accessory :switch
      :active test-networks-enabled?}
     {:size :small
modified   src/status_im2/contexts/profile/settings/events.cljs
@@ -61,6 +61,11 @@
  (fn [_ [value]]
    {:fx [[:dispatch [:profile.settings/profile-update :test-networks-enabled? (boolean value) {}]]]}))
 
+(rf/reg-event-fx :profile.settings/toggle-test-networks
+ (fn [{:keys [db]}]
+   (let [value (get-in db [:profile/profile :test-networks-enabled?])]
+     {:fx [[:dispatch [:profile.settings/profile-update :test-networks-enabled? (not value) {}]]]})))
```

**Comment by J-Son89** [comment]
@ajayesivan, why don't we make logout automatic here?

**Review by vkjr — APPROVED** [review]

**Comment by ajayesivan** [comment]
> @ajayesivan, why don't we make logout automatic here?

@J-Son89 On desktop they are not logging out. I mentioned it as a test note to confirm the toggle is working properly.

**Comment by J-Son89** [comment]
> > @ajayesivan, why don't we make logout automatic here?
> 
> @J-Son89 On desktop they are not logging out. I mentioned it as a test note to confirm the toggle is working properly.

okay that's fine for now - Desktop has setup it up that the login initialisation calls are repeated (at least for wallet) so that the tokens etc are on testnet. It might be good for us even to prompt/alert the user that they should logout/login again to avoid confusion among ourselves

**Review by ilmotta — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
hi, @ajayesivanThank you for the PR. Currently, I don't have any issues. I just have a few questions to ensure it should work so for now

### ISSUE 1:  ETH assets unavailable when test mode activated with mainnet or deactivated with goerli network enabled

### Steps:
1. Activate "test mode" and "Mainnet network" or deactivate "test mode" and  activate "Gerli network" 
image
2. go to the wallet
3. Open eth account

### Actual result:
Upon opening ETH account, the assets are not visible
image

### Expected result:
Potential solution:
- if testnet mode is activated, then the tests assets always should be visible
- if testnet mode is deactivated,  then the mainnet assets always should be visible

**Comment by VolodLytvynenko** [comment]
@ajayesivan Is this an [issue](https://github.com/status-im/status-mobile/pull/17879#issuecomment-1850387588) or a temporary behavior until the wallet setting screen, as mentioned by @J-Son89 in [this comment](https://github.com/status-im/status-mobile/pull/17879#issuecomment-1808418724), gets implemented?

**Comment by J-Son89** [comment]
ISSUE 1: ETH assets unavailable when test mode activated with mainnet or deactivated with goerli network enabled

This pr just adds the testnet network toggle that will make that change in Status-Go. There are a few minor adjustments we then need to make in Wallet once we have this pr merged so we will handle that in a separate issue/pr which someone from the wallet team can take 👍

**Comment by VolodLytvynenko** [comment]
@ajayesivan, you mentioned in [this comment](https://github.com/status-im/status-mobile/pull/17879#issue-1988441883) that 'This option is required to test curated communities in status.test fleet.'


 when I visit https://community-dapp-git-review-status-im-web.vercel.app/votes testnet (Optimism Goerli) to view the 'status.test' fleet communities, I don't see any community listed there. However, one community is fetched in the Status app in the Curated Communities section when the 'test mode' is activated. I'm a bit confused about what kind of community is being fetched into our Status app?

### Actual result:
No communities are shown in the https://community-dapp-git-review-status-im-web.vercel.app/votes  but one community is fetched in Status app in discovery page
image


### Expected result:
If no communities are shown in https://community-dapp-git-review-status-im-web.vercel.app/votes , then no communities are fetched to Curated communities page

**Comment by ajayesivan** [comment]
Hi @VolodLytvynenko,
As @J-Son89's mentioned, the goal of this pull request is to introduce the option to toggle the Test Networks state. This PR does not include any additional fixes for the wallet or curated communities. Instead, it's a preliminary step to facilitate upcoming fixes.

Concerning curated communities, to retrieve test network communities, we must enable both this toggle and switch to the `status.test` fleet. However, additional changes are still required to fully address the [curated communities issue](https://github.com/status-im/status-mobile/issues/17852).

**Comment by VolodLytvynenko** [comment]
hi @ajayesivan . Thank you for the clarification. No additional issues from my side. PR is ready to be merged
